File: 2582-pass-the-pillow/2582-pass-the-pillow.cpp
class Solution {
public:
    int passThePillow(int n, int time) {
        int t=1;
        int count=0;
        
        if(time<n)
        {
            return time+1;
        }
        
        for(int i=1; i<=time;i++)
        {
            if(t==n)
            {
                count=1;
            }
            if(t==1)
            {
                count=0;
            }
            if(count==1)
            {
                t--;
            }
            if(count==0)
            {
                t++;
            }
        }
        return t;
    }
};

File: 2582-pass-the-pillow/2582-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
        int pass=1;
        int result=1;
        
        for(int i=1; i<=time;i++)
        {
            result+=pass;
            if(result==1 || result==n)
            {
                pass*=-1;
            }
        }
        return result;
    }
}